ci/filter-tasks.js

```diff
@@ -0,0 +1,156 @@
+            return makeOptions.tasks;
+        }
+    }
+    catch (err) {
+        // If unable to reach rest client, build everything.
+        return makeOptions.tasks;
+    }
+    var packageMap = {};
+    packageInfo.result.value.forEach(package => {
+        if (package.name && package.versions) {
+            var packageName = package.name.slice('Mseng.MS.TF.DistributedTask.Tasks.'.length);
+            var packageVersion = package.versions[0].version;
+            package.versions.forEach(versionInfo => {
+                if (versionInfo.isLatest) {
+                    packageVersion = versionInfo.version;
+                }
+            });
+            packageMap[packageName] = packageVersion;
+        }
+    });
+
+    return makeOptions.tasks.filter(function (taskName) {
+        if (taskName in packageMap) {
+            var packageVersion = packageMap[taskName].split('.');
+            var packageMajorVersion = packageVersion[0];
+            var packageMinorVersion = packageVersion[1];
+            var packagePatchVersion = packageVersion[2];
+
+            var taskJsonPath = path.join(__dirname, '..', 'Tasks' , taskName, 'task.json')
+            var taskJson = JSON.parse(fs.readFileSync(taskJsonPath).toString());
+            var version = taskJson.version;
+            if (packageMajorVersion != version.Major || packageMinorVersion != version.Minor || packagePatchVersion != version.Patch) {
+                // If versions are not the same, want to build this package
+                return true;
+            }
+        }
+        return false;
+    });
+}
+
+var getTasksUsingChangedCommon = function(commonFilePaths) {
+    // Takes in an array of filepaths that have been changed in the Tasks/Common folder, returns any tasks that could be affected.
+    if (commonFilePaths.length == 0) {
+        return [];
+    }
+    changedTasks = [];
+    makeOptions.tasks.forEach(taskName => {
+        var makeJsonPath = path.join(__dirname, '..', 'Tasks' , taskName, 'make.json')
+        if (fs.existsSync(makeJsonPath)){
+            var makeJson = JSON.parse(fs.readFileSync(makeJsonPath).toString());
+            if (makeJson.common) {
+                makeJson.common.forEach(commonModule => {
+                    if (commonFilePaths.includes(commonModule.module.toLowerCase()) && !changedTasks.includes(taskName)) {
+                        changedTasks.push(taskName);
+                    }
+                });
+            }
+        }
+    });
+    return changedTasks;
+}
+
+var getTasksToBuildFromDiff = function() {
+    // Takes in a git source branch, diffs it with master, and returns a list of tasks that could have been affected by the changes.
+    var sourceBranch = process.env['SYSTEM_PULLREQUEST_SOURCEBRANCH'];
+    var prId = process.env['SYSTEM_PULLREQUEST_PULLREQUESTID'];
+    var commonChanges = [];
+    var toBeBuilt = [];
+    // Check if its from a fork
+    try {
+        if (sourceBranch.contains(':')) {
```

Comment:
Is there a reliable way to do this?

Reply:
I couldn't find a better way initially, I'll keep looking though. I agree this is non-ideal and a little hacky.

Reply:
Update - realized I could simplify a little bit, I'm more ok with where it is now because it only relies on finding the ':' to cut off a fork name. Now this will work even if only the second half is passed in all cases.